server/sonar-db-dao/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -339,4 +339,8 @@ public void setPrivateForRootComponentUuid(DbSession session, String projectUuid
   public void delete(DbSession session, long componentId) {
     mapper(session).delete(componentId);
   }
+
+  public List<ComponentDto> selectComponentKeysHavingIssuesToMerge(DbSession dbSession, String mergeBranchUuid) {
```

Comment:
At some point, maybe even in this POC, this should be optimized to not load entire `ComponentDto`, only the two columns in the query result: key and uuid. 
@dbmeneses Let me know if you think this should be done now.

Reply:
Nevermind, I'll just do this now, as the performance of this feature is critical for the MMF.

Reply:
Yes, I think we should do it now